command/server/listener.go

```diff
@@ -140,6 +141,34 @@ PASSPHRASECORRECT:
 		if err != nil {
 			return nil, nil, nil, errwrap.Wrapf("invalid value for 'tls_cipher_suites': {{err}}", err)
 		}
+
+		// HTTP/2 with TLS 1.2 blacklists several cipher suites.
+		// https://tools.ietf.org/html/rfc7540#appendix-A
+		//
+		// Since the CLI (net/http) automatically uses HTTP/2 with TLS 1.2,
+		// we check here if all or some specified cipher suites are blacklisted.
+		badCiphersCount := 0
+		badCiphers := []string{}
+		for _, cipher := range ciphers {
+			if isBadCipher(cipher) {
+				badCiphersCount++
+
+				// Get the name of the current cipher.
+				cipherStr, err := tlsutil.GetCipherName(cipher)
+				if err != nil {
+					return nil, nil, nil, errwrap.Wrapf("invalid value for 'tls_cipher_suites': {{err}}", err)
+				}
+				badCiphers = append(badCiphers, cipherStr)
+			}
+		}
+		if badCiphersCount == len(ciphers) {
+			return nil, nil, nil, errors.New("all defined cipher suites by 'tls_cipher_suites' are blacklisted by the HTTP/2 specification")
```

Comment:
I'm wavering on whether this should be an error or a warning. If their client is also appropriately configured things could still work even if it's out of spec. I'm leaning towards this being a warning as well, WDYT?

Reply:
I agree. I think some people will be frustrated if Vault will not start because of this and since HTTP/2 is still new for many enterprise companies it might be worth to stick to a warning instead of a hard failure.